### PR TITLE
Add basic support for ICMPv6

### PR DESCRIPTION
**Pull request by dlrobertson**
- Add packet structures for icmpv6
 - Add Tests
   - Add unit tests for icmpv6 checksum calculation
   - Update packetdump to handle icmpv6 packets

The implementation of `icmpv6` was heavily based on the implementation of `icmp`. Comments and critiques are very much welcomed.

### CODE REVIEW

**Comment by dlrobertson** [comment]
```
test pnettest::layer2_timeouts ... thread '<unnamed>' panicked at 'layer2_timeouts: should have exceeded timeout (66/46)', src/pnettest.rs:451:20
note: Run with `RUST_BACKTRACE=1` for a backtrace.
FAILED
test pnettest::layer3_ipv4 ... ok
test pnettest::layer4_ipv4 ... ok
test util::mac_addr_from_str ... ok
test util::str_from_mac_addr ... ok
test util::tests::sum_be_words_different_skipwords ... ok
failures:
---- pnettest::layer2_timeouts stdout ----
	thread 'pnettest::layer2_timeouts' panicked at 'assertion failed: res.join().is_ok()', src/pnettest.rs:470:4
failures:
    pnettest::layer2_timeouts
test result: FAILED. 41 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
error: test failed, to rerun pass '--lib'
make: *** [travis_script] Error 101
The command "make travis_script" exited with 2.
Done. Your build exited with 1.
```

Didn't see anything in `layer2_timeouts` related to this PR. Is this a known intermittent failure?

File: src/packet/ip.rs [hunk 1]
```diff
@@ -198,8 +198,8 @@ pub mod IpNextHeaderProtocols {
     /// SKIP
     pub const Skip: IpNextHeaderProtocol = IpNextHeaderProtocol(57);
 
-    /// ICMP for IPv6 [RFC2460]
-    pub const Ipv6Icmp: IpNextHeaderProtocol = IpNextHeaderProtocol(58);
+    /// ICMPv6 [RFC4443]
+    pub const Icmpv6: IpNextHeaderProtocol = IpNextHeaderProtocol(58);
```
Comment (mrmonday) [inline src/packet/ip.rs]: Please re-add `Ipv6Icmp` as `#[deprecated]` - this is a breaking API change.

**Comment by mrmonday** [comment]
Other than my one small comment, this looks good, thanks!

I haven't checked the constants - I'll trust they match the spec.

That test is a little sporadic, I wouldn't worry about it.

File: src/packet/ip.rs (hunk 1 above)
Comment (dlrobertson) [inline src/packet/ip.rs]: Good catch. Updated.

**Comment by dlrobertson** [comment]
> I haven't checked the constants - I'll trust they match the spec.

I only added the message types in [RFC 4443 section 2.1](https://tools.ietf.org/html/rfc4443#section-2.1). I'll be adding others shortly

> That test is a little sporadic, I wouldn't worry about it.

Cool, sounds good.

**Comment by mrmonday** [comment]
Good stuff, thanks!

**Pull request merged**